Assets/Scripts/Construction/PlayerConstructionHandler.cs

```diff
@@ -128,6 +136,19 @@ private Vector3 SnapToGrid(Vector3 vec, bool centerX, bool centerZ)
             return null;
         }
 
+        private void IsPlacementPositionValid()
+        {
+            Ray objectRay = new Ray(constructionPreview.transform.position, Vector3.down);
+            bool objectRaySuccess = Physics.Raycast(objectRay, out RaycastHit objectRayHit, 1f, buildTargetingCollisionMask);
+            if (objectRaySuccess && objectRayHit.transform != null && 
+                (objectRayHit.transform.gameObject.name == "Floor element" || objectRayHit.transform.gameObject.name == "Terrain")) // should be replaced later by a standard floor element prefab
```

Comment:
Tag it with TEMP keyword so we don't miss it (when searching via Ctrl+Shift+F)
```suggestion
                (objectRayHit.transform.gameObject.name == "Floor element" || objectRayHit.transform.gameObject.name == "Terrain")) // TEMP: should be replaced later by a standard floor element prefab
```